salt/modules/cp.py

```diff
@@ -8,6 +8,9 @@
 import os
 import logging
 import fnmatch
+import sys
+reload(sys)
+sys.setdefaultencoding('utf-8')
```

Comment:
Yeah. The example above is for a Windows specific module... so there would need to be a `salt.utils.is_windows()` check as well.

The real fix here, for Windows at least, is the eventual Python 3 migration. The Unicode stuff just works. These are stop gap measures that only work in the majority of cases. You're right, it may not handle non-UTF-8 character sets, but this seemed to handle most of things I was testing with...